core/trino-main/src/main/java/io/trino/operator/join/JoinHashSupplier.java

```diff
@@ -79,7 +83,20 @@ public JoinHashSupplier(
             positionLinksFactoryBuilder = ArrayPositionLinks.builder(addresses.size());
         }
 
+        long additionalPagesRetainedSizeInBytes = 0;
+        // Currently PageIndex is retained by the HashBuilderOperator after JoinHash is created.
+        // JoinHash has to account for all data structures retained by PageIndex as after JoinHash is created memory retained by PageIndex is no longer accounted for.
```

Comment:
> for all data structures retained by PageIndex as after JoinHash

Where `PagesIndex` is still referenced? Can we simplify it (pull retained memory computations or use memory context) rather than feeding back special parameters?

Reply:
Discussed offline. Decided to set `PageIndex` to `null` once it is no longer needed.